kratos/linear_solvers/fallback_linear_solver.h

```diff
@@ -0,0 +1,897 @@
+#include "factories/linear_solver_factory.h"
+
+namespace Kratos
+{
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/**
+ * @class FallbackLinearSolver
+ * @ingroup KratosCore
+ * @brief This is a simple linear solver that tries to solve the system using a list of solvers in order, until one of them succeeds.
+ * @details Stores a vector of solvers and tries to solve the system using the first one. If the solver fails, it tries the next one, and so on.
+ * @tparam TSparseSpaceType which specify type of the unknowns, coefficients, sparse matrix, vector of unknowns, right hand side vector and their respective operators.
+ * @tparam TDenseMatrixType which specify type of the matrices used as temporary matrices or multi solve unknowns and right hand sides and their operators.
+ * @tparam TReordererType which specify type of the Orderer that performs the reordering of matrix to optimize the solution.
+ * @see SparseSpace
+ * @see DenseSpace
+ * @see Reorderer
+ * @author Vicente Mataix Ferrandiz
+ */
+template<class TSparseSpaceType, class TDenseSpaceType, class TReordererType = Reorderer<TSparseSpaceType, TDenseSpaceType>>
+class FallbackLinearSolver
+    : public LinearSolver<TSparseSpaceType, TDenseSpaceType, TReordererType> {
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of LinearSolver
+    KRATOS_CLASS_POINTER_DEFINITION(FallbackLinearSolver);
+
+    /// Definition of the base class
+    using BaseType = LinearSolver<TSparseSpaceType, TDenseSpaceType, TReordererType>;
+
+    /// Definition of the base type pointer
+    using LinearSolverPointer = typename BaseType::Pointer;
+
+    /// The definition of the linear solver factory
+    using LinearSolverFactoryType = LinearSolverFactory<TSparseSpaceType, TDenseSpaceType>;
+
+    /// Type definition for sparse matrix
+    using SparseMatrixType = typename BaseType::SparseMatrixType;
+
+    /// Type definition for pointer to sparse matrix
+    using SparseMatrixPointerType = typename BaseType::SparseMatrixPointerType;
+
+    /// Type definition for vector
+    using VectorType = typename BaseType::VectorType;
+
+    /// Type definition for pointer to vector
+    using VectorPointerType = typename BaseType::VectorPointerType;
+
+    /// Type definition for dense matrix
+    using DenseMatrixType = typename BaseType::DenseMatrixType;
+
+    /// Type definition for dense vector
+    using DenseVectorType = typename BaseType::DenseVectorType;
+
+    /// Type definition for size
+    using SizeType = std::size_t;
+
+    /// Type definition for index
+    using IndexType = std::size_t;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /**
+     * @brief This is the default constructor
+     * @param ThisParameters The configuration parameters
+     */
+    FallbackLinearSolver(Parameters ThisParameters = Parameters(R"({})"))
```

Comment:
```suggestion
    explicit FallbackLinearSolver(Parameters ThisParameters = Parameters(R"({})"))
```

Reply:
btw why the default arguments? I thought we didn't do those.

Reply:
Yes, we do, look, many things use it, but in this case is a bit unnecessary